x-pack/plugins/enterprise_search/public/applications/app_search/components/documents/search_experience/build_search_ui_config.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Schema } from '../../../../shared/types';
+
+export const buildSearchUIConfig = (apiConnector: object, schema: Schema) => {
+  return {
+    alwaysSearchOnInitialLoad: true,
+    apiConnector,
+    trackUrlState: false,
+    initialState: {
+      sortDirection: 'desc',
+      sortField: 'id',
+    },
+    searchQuery: {
+      result_fields: Object.keys(schema || {}).reduce(
+        (acc: { [key: string]: object }, key: string) => {
+          acc[key] = {
+            snippet: {
+              size: 300,
+              fallback: true,
+            },
+            raw: {},
+          };
+          return acc;
+        },
+        {}
+      ),
```

Comment:
[NOT a hard change request, just me thinking out loud] Oh man, I always conceptually like the idea of `.reduce` but it feels so unintuitive to read haha. Or maybe that's just me.

I wonder if changing the var names to something a little less generic would make this a little easier to grok - what do you think?

```suggestion
      result_fields: Object.keys(schema || {}).reduce(
        (result_fields: { [field: string]: object }, field: string) => {
          result_fields[field] = {
            snippet: {
              size: 300,
              fallback: true,
            },
            raw: {},
          };
          return result_fields;
        },
        {}
      ),
```

Please ignore/don't commit this suggestion if you don't think it actually improves readability - I'm like 50/50 on this myself lol

Reply:
76d4fb6423ee91d93c3854d884f8aae3751c54a6

Reply:
reduce is definitely hard to read. For me, personally, I don't mind it, *as long as* it's in a reasonably small unit with a unit test explaining what is supposed to happen.

Reply:
Yeah for sure. There's actually some code in one of my upcoming DocumentCreation PRs where I thought about using a .reduce instead of a .map, but it felt worse/less readable. I'll def flag it later and ask for a 2nd opinion 😄

Reply:
actually sorry for not looking more closely before, but I'm slightly confused by 76d4fb6 - did this actually change anything? var names seem the same as before